ctparse/time/rules.py

```diff
@@ -395,27 +395,67 @@ def ruleDDMMYYYY(ts, m):
                 day=int(m.match.group('day')))
 
 
-@rule(r'(?<!\d|\.)(?P<hour>(?&_hour))((:|uhr|h|\.)?'
-      r'(?P<minute>(?&_minute))?\s*(uhr|h)?)(?P<ampm>\s*[ap]\.?m\.?)?(?!\d)')
-def ruleHHMM(ts, m):
-    # hh [am|pm]
-    # hh:mm
-    # hhmm
-    t = Time(hour=int(m.match.group('hour')),
-             minute=int(m.match.group('minute') or 0))
-    if m.match.group('ampm') is None:
+def _is_valid_military_time(ts, t):
+    t_year = t.hour * 100 + t.minute
+    # Military times (i.e. no separator) are notriously difficult to
+    # distinguish from yyyy; these are some heuristics to avoid an abundance
+    # of false positives for hhmm
+    #
+    # If hhmm is the current year -> assume it is a year
+    if t_year == ts.year:
+        return False
+    # If hhmm is the year in 3 month from now -> same, prefer year
+    if t_year == (ts + relativedelta(months=3)).year:
+        return False
+    # If the minutes is not a multiple of 5 prefer year (flight times are
+    # always multiples of 5 minutes)
+    if t.minute % 5:
+        return False
+    return True
+
+
+def _maybe_apply_am_pm(t, ampm_match):
+    if ampm_match is None:
         return t
-    elif m.match.group('ampm').lower().startswith('a') and t.hour <= 12:
+    elif ampm_match.lower().startswith('a') and t.hour <= 12:
         return t
-    elif m.match.group('ampm').lower().startswith('p') and t.hour < 12:
-        return Time(hour=t.hour+12, minute=t.minute)
+    elif ampm_match.lower().startswith('p') and t.hour < 12:
+        return Time(hour=t.hour + 12, minute=t.minute)
     else:
-        # the case m.match.group('ampm').startswith('a') and t.hour >
+        # the case ampm_match.startswith('a') and t.hour >
         # 12 (e.g. 13:30am) makes no sense, lets ignore the ampm
         # likewise if hour >= 12 no 'pm' action is needed
         return t
 
 
+@rule(r'(?<!\d|\.)(?P<hour>(?&_hour))(?P<minute>(?&_minute))'  # match hhmm
```

Comment:
`&hour` matches single digits as well - i.e. `123` would go as `1:23`...this is probably not what we want here. I would suggest to just explicetly put this regex here `r'(?:[01]\d)|(?:2[0-3])'`